### ISSUE TITLE
Deterministic validation set

### ISSUE THREAD

**Issue by mone27** [mone27/meteo_imp#13]
Currently the validation set can potentially be non-deterministic, if the var_sel and get lengths are generators. This is good for the training set, but for the validation set can be an issue because metric between training iterations are not the same.
Possible solutions are:
- have the variable selection and the gap len in the input of the pipeline, thus making the pipeline completely deterministic
- initialize the random state of the random generator (with the item id?) always to the same value in the validation pipeline
